core/valor_core/utilities.py

```diff
@@ -534,56 +573,84 @@ def _convert_groundtruth_or_prediction_to_dataframe(
+                    id_ = (
+                        str(ann_id) + str(i) + str(j) + str(k)
+                    )  # we use indices here, rather than a hash() so that the IDs are sequential. this prevents randomness when two predictions share the same score
+                    label_key = label.key
+                    label_value = label.value
+                    label_score = label.score
+                    label_id = hash(label_key + label_value)
+
+                    # only include scores for predictions
+                    if isinstance(obj, schemas.Prediction):
+                        output.append(
+                            {
+                                "datum_uid": datum_uid,
+                                "datum_id": datum_id,
+                                "datum_metadata": datum_metadata,
+                                "annotation_id": ann_id,
+                                "annotation_metadata": ann_metadata,
+                                "bounding_box": ann_bbox,
+                                "raster": ann_raster,
+                                "embedding": ann_embeding,
+                                "polygon": ann_polygon,
+                                "is_instance": ann_is_instance,
+                                "text": ann_text,
+                                "context_list": ann_context_list,
+                                "label_key": label_key,
+                                "label_value": label_value,
+                                "score": label_score,
+                                "label_id": label_id,
+                                "id": id_,
+                            }
+                        )
+                    else:
+                        output.append(
+                            {
+                                "datum_uid": datum_uid,
+                                "datum_id": datum_id,
+                                "datum_metadata": datum_metadata,
+                                "annotation_id": ann_id,
+                                "annotation_metadata": ann_metadata,
+                                "bounding_box": ann_bbox,
+                                "raster": ann_raster,
+                                "embedding": ann_embeding,
+                                "polygon": ann_polygon,
+                                "is_instance": ann_is_instance,
+                                "text": ann_text,
+                                "context_list": ann_context_list,
+                                "label_key": label_key,
+                                "label_value": label_value,
+                                "label_id": label_id,
+                                "id": id_,
+                            }
+                        )
+            else:
                 id_ = (
-                    str(ann_id) + str(i) + str(j) + str(k)
+                    str(ann_id) + str(i) + str(j) + "0"
```

Comment:
I don't have any testing for this id_ because none of my tests actually use this dataframe structure for the data.

Reply:
I'm fine to remove all the text gen stuff from these utilities if that's what you guys think is best, as some of those additions are untested as I am not using the same dataframe structure